apps/api/src/integration/plugins/slack.integration.ts

```diff
@@ -10,28 +12,73 @@ import {
 import { App } from '@slack/bolt'
 import { BaseIntegration } from './base.integration'
 import { PrismaService } from '@/prisma/prisma.service'
-import { decryptMetadata, makeTimedRequest } from '@/common/util'
+import { makeTimedRequest } from '@/common/util'
 import { InternalServerErrorException } from '@nestjs/common'
 
 export class SlackIntegration extends BaseIntegration {
-  private readonly app: App
+  private app: App
 
   constructor(prisma: PrismaService) {
     super(IntegrationType.SLACK, prisma)
+  }
 
-    if (!this.app && this.integration) {
-      const metadata = decryptMetadata<SlackIntegrationMetadata>(
-        this.integration.metadata
-      )
-      this.app = new App({
-        token: metadata.botToken,
-        signingSecret: metadata.signingSecret
+  public async init(
+    _privateKey: Project['privateKey'],
+    eventId: Event['id']
+  ): Promise<void> {
+    this.logger.log(`Initializing Slack Integration...`)
+
+    const integration = this.getIntegration<SlackIntegrationMetadata>()
+
+    try {
+      const { id: integrationRunId } = await this.registerIntegrationRun({
+        eventId: eventId,
+        integrationId: integration.id,
+        title: 'Posting message to Slack'
       })
+
+      const block = [
+        {
+          type: 'header',
+          text: {
+            type: 'plain_text',
+            text: '🥁 Keyshade is now configured with this channel!',
+            emoji: true
+          }
+        }
+      ]
+
+      this.app = await this.getSlackApp()
+
+      const { response, duration } = await makeTimedRequest(() =>
+        this.app.client.chat.postMessage({
+          channel: integration.metadata.channelId,
+          blocks: block,
+          text: 'Integration initialized'
+        })
+      )
+
+      await this.markIntegrationRunAsFinished(
+        integrationRunId,
+        response.ok
+          ? IntegrationRunStatus.SUCCESS
+          : IntegrationRunStatus.FAILED,
+        duration,
+        response.message.text
+      )
+
+      if (!response.ok) {
+        this.logger.error(
+          `Failed to integrate Slack: ${response.status} - ${response.statusText}`
+        )
+      } else {
+        this.logger.log(`Slack integration initialized in ${duration}ms`)
+      }
+    } catch (error) {
+      this.logger.error(`Failed to integrate Slack: ${error}`)
```

Comment:
In case of failure aswell, mark the status of the run using `this.markIntegrationRunAsFinished` and set the status accordingly